### PR TITLE
feat: support topologySpreadConstraints

### PR DESCRIPTION
**Pull request by smauermann**
This PR adds support for configuring topologySpreadConstraints on the following components:
- chartmuseum
- core
- exporter
- jobservice
- nginx
- notary.server
- notary.signer
- portal
- registry
- trivy

Solves issue https://github.com/goharbor/harbor-helm/issues/1294.

### CODE REVIEW

**Comment by robinkb** [comment]
@ywk253100 What is the status of this PR? Can we please have a review? I need this feature as well. It's very difficult to reliably scale Harbor across availability zones without this.

Sorry if I tagged the wrong person, I couldn't find  the maintainer of this project after searching for a while.

File: templates/chartmuseum/chartmuseum-dpl.yaml [hunk 1]
```diff
@@ -49,6 +49,16 @@ spec:
         {{- toYaml . | nindent 8 }}
       {{- end }}
       automountServiceAccountToken: {{ .Values.chartmuseum.automountServiceAccountToken | default false }}
+{{- with .Values.chartmuseum.topologySpreadConstraints}}
+      topologySpreadconstraints:
```
Comment (zyyw) [inline templates/chartmuseum/chartmuseum-dpl.yaml:53]: please change `topologySpreadconstraints` to `topologySpreadConstraints`

**Comment by zyyw** [comment]
@smauermann please help to resolve the comment.

**Comment by smauermann** [comment]
@zyyw Will get to it today!

**Comment by robinkb** [comment]
Hi @smauermann, do you have time to pick this up shortly? Maybe I can pick up the PR instead?

**Comment by fplantinga-guida** [comment]
Any updates on this?

**Comment by smauermann** [comment]
I have pushed the requested changes.

**Comment by smauermann** [comment]
@zyyw would you mind unblocking CI?

**Comment by dioguerra** [comment]
I'm picking this up again,

**Pull request closed without merging**